wgsl/index.bs

```diff
@@ -4180,6 +4180,242 @@ TODO: deduplicate these tables
+textureSampleBias(texture : texture_depth_2d_array,           sampler : sampler, coords : vec2<f32>, array_index : u32, bias : f32, offset : vec2<i32>) -> f32
+textureSampleBias(texture : texture_depth_cube,               sampler : sampler, coords : vec3<f32>,                    bias : f32                    ) -> f32
+textureSampleBias(texture : texture_depth_cube_array,         sampler : sampler, coords : vec3<f32>, array_index : u32, bias : f32                    ) -> f32
+```
+
+**Parameters:**
+
+<table class='data'>
+  <tr><td>`texture`<td>
+  The [sampled](#sampled-texture-type) or [depth](#texture-depth) texture to
+  sample.
+  <tr><td>`sampler`<td>
+  The [sampler type](#sampler-type).
+  <tr><td>`coords`<td>
+  The texture coordinates used for sampling.
+  <tr><td>`array_index`<td>
+  The 0-based texture array index to sample.
+  <tr><td>`bias`<td>
+  The bias to apply to the mip level before sampling.
+  `bias` must be between `-16.0` and `15.99`.
+  <tr><td>`offset`<td>
+  The optional texel offset applied to the unnormalized texture coordinate
+  before sampling the texture. This offset is applied before applying any
+  texture wrapping modes. `offset` values may range from `[-8..7]` in all
+  dimensions. Values outside of this range will be clamped to the nearest limit.
+</table>
+
+**Returns:**
+
+The sampled value.
+
+
+### `textureSampleLevel` ### {#texturesamplelevel}
+
+Samples a texture using an explicit mip level.
+
+```rust
+textureSampleLevel(texture : texture_sampled_2d<type>,         sampler : sampler, coords : vec2<f32>,                    level : f32                    ) -> <type>
+textureSampleLevel(texture : texture_sampled_2d<type>,         sampler : sampler, coords : vec2<f32>,                    level : f32, offset : vec2<i32>) -> <type>
+textureSampleLevel(texture : texture_sampled_2d_array<type>,   sampler : sampler, coords : vec2<f32>, array_index : u32, level : f32                    ) -> <type>
+textureSampleLevel(texture : texture_sampled_2d_array<type>,   sampler : sampler, coords : vec2<f32>, array_index : u32, level : f32, offset : vec2<i32>) -> <type>
+textureSampleLevel(texture : texture_sampled_3d<type>,         sampler : sampler, coords : vec3<f32>,                    level : f32                    ) -> <type>
+textureSampleLevel(texture : texture_sampled_3d<type>,         sampler : sampler, coords : vec3<f32>,                    level : f32, offset : vec3<i32>) -> <type>
+textureSampleLevel(texture : texture_sampled_cube<type>,       sampler : sampler, coords : vec3<f32>,                    level : f32                    ) -> <type>
+textureSampleLevel(texture : texture_sampled_cube_array<type>, sampler : sampler, coords : vec3<f32>, array_index : u32, level : f32                    ) -> <type>
+textureSampleLevel(texture : texture_depth_2d,                 sampler : sampler, coords : vec2<f32>,                    level : f32                    ) -> f32
```

Comment:
I don't think we get linear interpolation for any of the depth formats. Therefore, the level should probably be `u32`

Reply:
Before I change this, do you have a reference that I can look up for this?

Reply:
It's in the same table I linked in another comment - https://gpuweb.github.io/gpuweb/#texture-format-caps

Reply:
Thanks. Done.